Automattic/jetpack
# Search: Fix irrelevant widgets in overlay sidebar on auto config

Fixes #24402 

#### Changes proposed in this Pull Request:
Changes the name of the overlay sidebar to walk around the WP widget auto-adding mechanism  - WP migrates/adds some default widgets to sidebars with keyword `sidebar` in it. To walk around that, we changed `jetpack-instant-search-sidebar` to `jetpack-instant-search-side-bar`. In order to maintain the compatibility for old search sidebar, we hooks the option value and does a mapping.

It also removes some hacky code to stop these widgets to be added when switching themes.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:

* Create a JN site from [this link](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-search=master&nojetpack) with only Jetpack Beta Tester plugin and Search plugin on bleeding edge
* Purchase a search subscription
* Ensure Jetpack Search sidebar is auto configured with categories and tags filter (and some irrelevant widgets)
* Switch Search to branch `fix/irrelevant-widgets-in-overlay-sidebar`
* Ensure Jetpack Search sidebar still works
* Ensure you could edit the content of the sidebar
-------

* Create another JN site from [this link](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack-search=fix/irrelevant-widgets-in-overlay-sidebar&nojetpack) with only Jetpack Beta Tester plugin and Search plugin on branch `fix/irrelevant-widgets-in-overlay-sidebar`
* Purchase a search subscription
* Ensure only Jetpack Search widget is added to overlay sidebar with only categories and tags filter (no irrelevant widgets)
